src/javascript/binary/pages/trade/event.js

```diff
@@ -237,8 +237,7 @@ var TradingEvents = (function () {
         var amountElement = document.getElementById('amount');
         if (amountElement) {
             amountElement.addEventListener('input', debounce( function(e) {
-                var amount_val = parseFloat(e.target.value);
-                if (e.target.value % 1 !== 0 && ((+amount_val).toFixed(10)).replace(/^-?\d*\.?|0+$/g, '').length>2) {
+                if (e.target.value % 1 !== 0 && isStandardFloat(parseFloat(e.target.value))) {
```

Comment:
as you are checking this `e.target.value % 1 !== 0` everywhere why not move it to `isStandardFloat`